lib/grpc/GrpcServer.ts

```diff
@@ -0,0 +1,49 @@
+import grpc, { Server } from 'grpc';
+import Logger from '../Logger';
+import errors from './errors';
+import GrpcService from './GrpcService';
+import Service from '../service/Service';
+import { WalliService } from '../proto/wallirpc_grpc_pb';
+import assert from 'assert';
+
+class GrpcServer {
+  private server: Server;
+  private logger: Logger;
+
+  constructor(logger: Logger, service: Service) {
+    this.server = new grpc.Server();
+    this.logger = logger;
+
+    const grpcService = new GrpcService(logger, service);
+    this.server.addService(WalliService, {
+      getInfo: grpcService.getInfo,
+    });
+  }
+
+  public listen = async (port: number, host: string) => {
+    return new Promise((resolve, reject) => {
```

Comment:
Why are you returning a `Promise` here? When declaring a function as `async` you don't have to do this. `return` and `throw` will have the same effect as `resolve` and `reject`.